datajoint/fetch.py

```diff
@@ -20,46 +25,57 @@ def prepare_attributes(relation, item):
         raise DataJointError("Index must be a slice, a tuple, a list, a string.")
     return item, attributes
 
-class FetchQuery:
+def copy_first(f):
+    @wraps(f)
+    def ret(*args, **kwargs):
+        args = list(args)
+        args[0] = args[0].__class__(args[0]) # call copy constructor
+        return f(*args, **kwargs)
 
-    def __init__(self, relation):
-        """
+    return ret
 
-        """
-        self.behavior = dict(
-            offset=0, limit=None, order_by=None, descending=False, as_dict=False, map=None
-        )
-        self._relation = relation
+class Fetch:
+    def __init__(self, relation):
+        if isinstance(relation, Fetch): # copy constructor
+            self.behavior = dict(relation.behavior)
+            self._relation = relation._relation
+        else:
+            self.behavior = dict(
+                offset=0, limit=None, order_by=None, as_dict=False
+            )
+            self._relation = relation
 
 
+    @copy_first
     def from_to(self, fro, to):
         self.behavior['offset'] = fro
         self.behavior['limit'] = to - fro
         return self
 
-    def order_by(self, order_by):
-        self.behavior['order_by'] = order_by
+    @copy_first
+    def order_by(self, *args):
+        if len(args) > 0:
+            self.behavior['order_by'] = self.behavior['order_by'] if self.behavior['order_by'] is not None else []
+            namepat = re.compile(r"\s*(?P<name>\w+).*")
+            for a in args: # remove duplicates
```

Comment:
Each method call creating separate `fetch` object is consistent with how SQLAlchemy handles their `Query` objects. I actually just checked SQLAlchemy and the query it generates, and it looks like it does not go about removing duplicates - it simply appends all statements into SQL. As for SQL, it's completely fine executing queries like `ORDER_BY name DESC, language ASC, name ASC`. As it sorts from last to first, this will cause the newer `sort_by('name ASC')` call to be effecitvely ignored.

One thing that SQLAlchemy supports that may be nice is to call `order_by(None)`. This returns a new query object with all `order_by` reset. Call to `order_by` without an argument is simply ignored in SQLAlchemy